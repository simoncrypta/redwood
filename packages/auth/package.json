{
  "name": "@redwoodjs/auth",
  "version": "0.25.1",
  "files": [
    "dist"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "license": "MIT",
  "devDependencies": {
    "@auth0/auth0-spa-js": "^1.7.0",
    "@supabase/supabase-js": "^1.3.3",
    "@types/netlify-identity-widget": "^1.4.1",
    "@types/react": "16.9.53",
    "firebase": "^7.14.5",
    "firebase-admin": "^9.1.1",
    "gotrue-js": "git://github.com/netlify/gotrue-js.git#28df09cfcac505feadcb1719714d2a9507cf68eb",
    "magic-sdk": "^2.5.0",
    "msal": "^1.4.1",
    "netlify-identity-widget": "1.9.1",
    "nhost-js-sdk": "^3.0.0-14",
    "react": "^16.13.1"
  },
  "bundlesize": [
    {
      "path": "./dist/AuthProvider.js",
      "maxSize": "10 kB",
      "compression": "none"
    },
    {
      "path": "./dist/authClients/*.js",
      "maxSize": "5 kB",
      "compression": "none"
    }
  ],
  "scripts": {
    "build": "yarn build:js && yarn build:types",
    "prepublishOnly": "yarn cross-env NODE_ENV=production yarn build",
    "build:js": "babel src -d dist --extensions \".js,.ts,.tsx\"",
    "build:types": "ttsc --build --verbose",
    "build:watch": "nodemon --watch src --ext \"js,ts,tsx\" --ignore dist --exec \"yarn build\"",
    "test": "jest",
    "test:watch": "yarn test --watch"
  },
  "gitHead": "c235e7d7186e5e258764699c0e0e1d5cc0fdd0b5"
}
