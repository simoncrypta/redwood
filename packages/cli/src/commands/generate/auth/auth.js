import { generateDepracatedHandler } from '../generateDepracatedHandler'

// ********
// Deprecated as of September 2020
// Use "setup" command
// ********

export const command = 'auth'
export const description = 'WARNING: deprecated. Use "yarn rw setup" command.'

export const handler = generateDepracatedHandler({
  newCommand: 'yarn rw setup auth ${provider}',
  docsLink: 'https://redwoodjs.com/reference/command-line-interface#setup',
})
