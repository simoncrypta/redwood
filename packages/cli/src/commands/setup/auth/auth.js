import fs from 'fs'
import path from 'path'

import execa from 'execa'
import Listr from 'listr'
import terminalLink from 'terminal-link'

import { resolveFile } from '@redwoodjs/internal'

import { getPaths, writeFilesTask } from 'src/lib'
import c from 'src/lib/colors'

const API_GRAPHQL_PATH = resolveFile(
  path.join(getPaths().api.functions, 'graphql')
)

const AUTH_PROVIDER_IMPORT = `import { AuthProvider } from '@redwoodjs/auth'`

const API_SRC_PATH = path.join(getPaths().api.src)
const TEMPLATES = fs
  .readdirSync(path.resolve(__dirname, 'templates'))
  .reduce((templates, file) => {
    if (file === 'auth.js.template') {
      return { ...templates, base: path.resolve(__dirname, 'templates', file) }
    } else {
      const provider = file.replace('.auth.js.template', '')
      return {
        ...templates,
        [provider]: path.resolve(__dirname, 'templates', file),
      }
    }
  }, {})

const OUTPUT_PATH = path.join(getPaths().api.lib, 'auth.js')
const WEB_SRC_INDEX_PATH = path.join(getPaths().web.src, 'index.js')

const SUPPORTED_PROVIDERS = fs
  .readdirSync(path.resolve(__dirname, 'providers'))
  .map((file) => path.basename(file, '.js'))
  .filter((file) => file !== 'README.md')

// returns the content of index.js with import statements added
const addWebImports = (content, imports) => {
  return `${AUTH_PROVIDER_IMPORT}\n` + imports.join('\n') + '\n' + content
}

// returns the content of index.js with init lines added
const addWebInit = (content, init) => {
  return content.replace(
    'const App = () => (',
    `${init}\n\nconst App = () => (`
  )
}

// returns the content of index.js with <AuthProvider> added
const addWebRender = (content, authProvider) => {
  const [_, indent, redwoodApolloProvider] = content.match(
    /(\s+)(<RedwoodApolloProvider>.*<\/RedwoodApolloProvider>)/s
  )

  const redwoodApolloProviderLines = redwoodApolloProvider
    .split('\n')
    .map((line) => {
      return '  ' + line
    })

  const customRenderOpen = (authProvider.render || []).reduce(
    (acc, component) => acc + indent + `<${component}>`,
    ''
  )

  const customRenderClose = (authProvider.render || []).reduce(
    (acc, component) => indent + `</${component}>` + acc,
    ''
  )

  const renderContent =
    customRenderOpen +
    indent +
    `<AuthProvider client={${authProvider.client}} type="${authProvider.type}">` +
    indent +
    redwoodApolloProviderLines.join('\n') +
    indent +
    `</AuthProvider>` +
    customRenderClose

  return content.replace(
    /\s+<RedwoodApolloProvider>.*<\/RedwoodApolloProvider>/s,
    renderContent
  )
}

// returns the content of index.js with <AuthProvider> updated
const updateWebRender = (content, authProvider) => {
  const renderContent = `<AuthProvider client={${authProvider.client}} type="${authProvider.type}">`
  return content.replace(/<AuthProvider client={.*} type=".*">/s, renderContent)
}

// returns the content of index.js without the old auth import
const removeOldWebImports = (content, imports) => {
  return content.replace(`${AUTH_PROVIDER_IMPORT}\n` + imports.join('\n'), '')
}

// returns the content of index.js without the old auth init
const removeOldWebInit = (content, init) => {
  return content.replace(init, '')
}

// returns content with old auth provider removes
const removeOldAuthProvider = async (content) => {
  // get the current auth provider
  const [_, currentAuthProvider] = content.match(
    /<AuthProvider client={.*} type="(.*)">/s
  )

  let oldAuthProvider
  try {
    oldAuthProvider = await import(`./providers/${currentAuthProvider}`)
  } catch (e) {
    throw new Error('Could not replace existing auth provider init')
  }

  content = removeOldWebImports(content, oldAuthProvider.config.imports)
  content = removeOldWebInit(content, oldAuthProvider.config.init)

  return content
}

// check to make sure AuthProvider doesn't exist
const checkAuthProviderExists = () => {
  const content = fs.readFileSync(WEB_SRC_INDEX_PATH).toString()

  if (content.includes(AUTH_PROVIDER_IMPORT)) {
    throw new Error(
      'Existing auth provider found.\nUse --force to override existing provider.'
    )
  }
}

// the files to create to support auth
export const files = (provider) => {
  const template = TEMPLATES[provider] ?? TEMPLATES.base
  return {
    [OUTPUT_PATH]: fs.readFileSync(template).toString(),
  }
}

// actually inserts the required config lines into index.js
export const addConfigToIndex = async (config, force) => {
  let content = fs.readFileSync(WEB_SRC_INDEX_PATH).toString()

  // update existing AuthProvider if --force else add new AuthProvider
  if (content.includes(AUTH_PROVIDER_IMPORT) && force) {
    content = await removeOldAuthProvider(content)
    content = updateWebRender(content, config.authProvider)
  } else {
    content = addWebRender(content, config.authProvider)
  }

  content = addWebImports(content, config.imports)
  content = addWebInit(content, config.init)

  fs.writeFileSync(WEB_SRC_INDEX_PATH, content)
}

export const addApiConfig = () => {
  let content = fs.readFileSync(API_GRAPHQL_PATH).toString()

  // default to an array to avoid destructure errors
  const [_, hasAuthImport] =
    content.match(/(import {.*} from 'src\/lib\/auth.*')/s) || []

  if (!hasAuthImport) {
    // add import statement
    content = content.replace(
      /^(.*services.*)$/m,
      `$1\n\nimport { getCurrentUser } from 'src/lib/auth'`
    )
    // add object to handler
    content = content.replace(
      /^(\s*)(schema: makeMergedSchema)(.*)$/m,
      `$1getCurrentUser,\n$1$2$3`
    )
    fs.writeFileSync(API_GRAPHQL_PATH, content)
  }
}

export const isProviderSupported = (provider) => {
  return SUPPORTED_PROVIDERS.indexOf(provider) !== -1
}

export const apiSrcDoesExist = () => {
  return fs.existsSync(API_SRC_PATH)
}

export const webIndexDoesExist = () => {
  return fs.existsSync(WEB_SRC_INDEX_PATH)
}

export const graphFunctionDoesExist = () => {
  return fs.existsSync(API_GRAPHQL_PATH)
}

export const command = 'auth <provider>'
export const description = 'Generate an auth configuration'
export const builder = (yargs) => {
  yargs
    .positional('provider', {
      choices: SUPPORTED_PROVIDERS,
      description: 'Auth provider to configure',
      type: 'string',
    })
    .option('force', {
      alias: 'f',
      default: false,
      description: 'Overwrite existing configuration',
      type: 'boolean',
    })
    .epilogue(
      `Also see the ${terminalLink(
        'Redwood CLI Reference',
        'https://redwoodjs.com/reference/command-line-interface#generate-auth'
      )}`
    )
}

export const handler = async ({ provider, force }) => {
  const providerData = await import(`./providers/${provider}`)

  const tasks = new Listr(
    [
      {
        title: 'Generating auth lib...',
        task: (_ctx, task) => {
          if (apiSrcDoesExist()) {
            return writeFilesTask(files(provider), { overwriteExisting: force })
          } else {
            task.skip('api/src not found, skipping')
          }
        },
      },
      {
        title: 'Adding auth config to web...',
        task: (_ctx, task) => {
          if (webIndexDoesExist()) {
            addConfigToIndex(providerData.config, force)
          } else {
            task.skip('web/src/index.js not found, skipping')
          }
        },
      },
      {
        title: 'Adding auth config to GraphQL API...',
        task: (_ctx, task) => {
          if (graphFunctionDoesExist()) {
            addApiConfig()
          } else {
            task.skip('GraphQL function not found, skipping')
          }
        },
      },
      {
        title: 'Adding required web packages...',
        task: async () => {
          if (!isProviderSupported(provider)) {
            throw new Error(`Unknown auth provider '${provider}'`)
          }
          await execa('yarn', [
            'workspace',
            'web',
            'add',
            ...providerData.webPackages,
            '@redwoodjs/auth',
          ])
        },
      },
      providerData.apiPackages.length > 0 && {
        title: 'Adding required api packages...',
        task: async () => {
          if (!isProviderSupported(provider)) {
            throw new Error(`Unknown auth provider '${provider}'`)
          }
          await execa('yarn', [
            'workspace',
            'api',
            'add',
            ...providerData.apiPackages,
          ])
        },
      },
      {
        title: 'Installing packages...',
        task: async () => {
          await execa('yarn', ['install'])
        },
      },
      {
        title: 'One more thing...',
        task: (_ctx, task) => {
          task.title = `One more thing...\n\n   ${providerData.notes.join(
            '\n   '
          )}\n`
        },
      },
    ].filter(Boolean),
    { collapse: false }
  )

  try {
    // Don't throw existing provider error when --force exists
    if (!force) {
      checkAuthProviderExists()
    }

    await tasks.run()
  } catch (e) {
    console.error(c.error(e.message))
    process.exit(e?.exitCode || 1)
  }
}
