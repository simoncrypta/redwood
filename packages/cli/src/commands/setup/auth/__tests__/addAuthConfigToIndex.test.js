import fs from 'fs'

import { addConfigToIndex } from '../auth'

jest.mock('src/lib', () => {
  const path = require('path')
  const __dirname = path.resolve()
  return {
    getPaths: () => ({
      api: { functions: '', src: '', lib: '' },
      web: {
        src: path.join(__dirname, '../create-redwood-app/template/web/src'),
      },
    }),
  }
})

// This function checks output matches
const writeFileSyncSpy = jest.fn((_, content) => {
  // Line breaks cause an issue on windows snapshots
  expect(content).toMatchSnapshot()
})

beforeEach(() => {
  jest.restoreAllMocks()
  jest.spyOn(fs, 'writeFileSync').mockImplementation(writeFileSyncSpy)
})

describe('Should add config lines to index.js', () => {
  it('Matches Auth0 Snapshot', async () => {
    const auth0Data = await import(`../providers/auth0`)
    await addConfigToIndex(auth0Data.config, false)
  })

  it('Matches Firebase Snapshot', async () => {
    const firebaseData = await import(`../providers/firebase`)
    await addConfigToIndex(firebaseData.config, false)
  })

  it('Matches netlify Snapshot', async () => {
    const netlifyData = await import(`../providers/netlify`)
    await addConfigToIndex(netlifyData.config, false)
  })

  it('Matches goTrue Snapshot', async () => {
    const goTrueData = await import(`../providers/goTrue`)
    await addConfigToIndex(goTrueData.config, false)
  })

  it('Matches azureActiveDirectory Snapshot', async () => {
    const azureActiveDirectoryData = await import(
      `../providers/azureActiveDirectory`
    )
    await addConfigToIndex(azureActiveDirectoryData.config, false)
  })

  it('Matches ethereum Snapshot', async () => {
    const ethereumData = await import(`../providers/ethereum`)
    await addConfigToIndex(ethereumData.config, false)
  })

  it('Matches magicLink Snapshot', async () => {
    const magicLinkData = await import(`../providers/magicLink`)
    await addConfigToIndex(magicLinkData.config, false)
  })

  it('Matches supabase Snapshot', async () => {
    const supabaseData = await import(`../providers/supabase`)
    await addConfigToIndex(supabaseData.config, false)
  })

  it('Matches nhost Snapshot', async () => {
    const nhostData = await import(`../providers/nhost`)
    await addConfigToIndex(nhostData.config, false)
  })
})
