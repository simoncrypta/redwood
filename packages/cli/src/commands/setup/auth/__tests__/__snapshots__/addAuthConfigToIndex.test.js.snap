// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should add config lines to index.js Matches Auth0 Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import { Auth0Client } from '@auth0/auth0-spa-js'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const auth0 = new Auth0Client({
  domain: process.env.AUTH0_DOMAIN,
  client_id: process.env.AUTH0_CLIENT_ID,
  redirect_uri: process.env.AUTH0_REDIRECT_URI,
  // ** NOTE ** Storing tokens in browser local storage provides persistence across page refreshes and browser tabs.
  // However, if an attacker can achieve running JavaScript in the SPA using a cross-site scripting (XSS) attack,
  // they can retrieve the tokens stored in local storage.
  // https://auth0.com/docs/libraries/auth0-spa-js#change-storage-options
  cacheLocation: 'localstorage',
  audience: process.env.AUTH0_AUDIENCE,
})

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={auth0} type=\\"auth0\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches Firebase Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import * as firebase from 'firebase/app'
import 'firebase/auth'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const firebaseClientConfig = {
  apiKey: process.env.FIREBASE_API_KEY,
  authDomain: process.env.FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.FIREBASE_DATABASE_URL,
  projectId: process.env.FIREBASE_PROJECT_ID,
  storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.FIREBASE_APP_ID,
}

const firebaseClient = ((config) => {
  firebase.initializeApp(config)
  return firebase
})(firebaseClientConfig)

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={firebaseClient} type=\\"firebase\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches azureActiveDirectory Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import { UserAgentApplication } from 'msal'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const azureActiveDirectoryClient = new UserAgentApplication({
    auth: {
      clientId: process.env.AZURE_ACTIVE_DIRECTORY_CLIENT_ID,
      authority: process.env.AZURE_ACTIVE_DIRECTORY_AUTHORITY,
      redirectUri: process.env.AZURE_ACTIVE_DIRECTORY_REDIRECT_URI,
      postLogoutRedirectUri: process.env.AZURE_ACTIVE_DIRECTORY_LOGOUT_REDIRECT_URI,
    },
  })

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={azureActiveDirectoryClient} type=\\"azureActiveDirectory\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches ethereum Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import EthereumAuthClient from '@oneclickdapp/ethereum-auth'
import { ApolloClient, InMemoryCache } from '@apollo/client'
import { FetchConfigProvider, useFetchConfig } from '@redwoodjs/web'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

let ethereum

  const ApolloInjector = ({ children }) => {
  const { uri, headers } = useFetchConfig()
  try {
    const graphQLClient = new ApolloClient({
      cache: new InMemoryCache(),
      uri,
      headers,
    })
    // Default option using Apollo Client
    const makeRequest = (mutation, variables) =>
      graphQLClient.mutate({
        mutation,
        variables,
      })

    // Alternative option using graphql-hooks
    // You'll also need to modify graphQLClient
    // const makeRequest = (query, variables) =>
    //   graphQLClient.request({
    //     query,
    //     variables,
    //   })

    ethereum = new EthereumAuthClient({
      makeRequest,
      debug: process.NODE_ENV === 'development',
    })
  } catch (e) {
    console.log(e)
  }
  return React.cloneElement(children, { client: ethereum })
}

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <FetchConfigProvider>
    <ApolloInjector>
    <AuthProvider client={ethereum} type=\\"ethereum\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
    </ApolloInjector>
    </FetchConfigProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches goTrue Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import GoTrue from 'gotrue-js'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const goTrueClient = new GoTrue({
  APIUrl: 'https://MYAPP.netlify.app/.netlify/identity',
  setCookie: true,
})

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={goTrueClient} type=\\"goTrue\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches magicLink Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import { Magic } from 'magic-sdk'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const m = new Magic(process.env.MAGICLINK_PUBLIC)

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={m} type=\\"magicLink\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches netlify Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import netlifyIdentity from 'netlify-identity-widget'
import { isBrowser } from '@redwoodjs/prerender/browserUtils'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

isBrowser && netlifyIdentity.init()

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={netlifyIdentity} type=\\"netlify\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches nhost Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import { createClient } from 'nhost-js-sdk'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const nhostClient = createClient({
  baseURL: process.env.NHOST_BACKEND_URL,
  autoLogin: false,
});


const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={nhostClient} type=\\"nhost\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;

exports[`Should add config lines to index.js Matches supabase Snapshot 1`] = `
"import { AuthProvider } from '@redwoodjs/auth'
import { createClient } from '@supabase/supabase-js'
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const supabaseClient = createClient(
    process.env.SUPABASE_URL,
    process.env.SUPABASE_KEY
  )

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <AuthProvider client={supabaseClient} type=\\"supabase\\">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </AuthProvider>
  </FatalErrorBoundary>
)

export default App
"
`;
